synapse/rest/admin/rooms.py

```diff
@@ -152,19 +156,23 @@ async def on_GET(
 
         delete_ids = self._pagination_handler.get_delete_ids_by_room(room_id)
         if delete_ids is None:
-            raise NotFoundError("No delete task for room_id '%s' found" % room_id)
+            delete_ids = []
 
         response = []
         for delete_id in delete_ids:
             delete = self._pagination_handler.get_delete_status(delete_id)
-            if delete:
+            if delete and delete.status != DeleteStatus.STATUS_WAIT_PURGE:
```

Comment:
Why don't we return scheduled purges in the admin API?

I suppose it's something about automatic vs manual purges that the admin would care about.

In any case, we should add the context in a comment here

Reply:
Indeed it's because it's not a manual action that was added by the API. Comment added.

Reply:
Continued in https://github.com/matrix-org/synapse/pull/15488#discussion_r1253323512